hw/hal/include/hal/hal_i2c_sim.h

```diff
@@ -0,0 +1,108 @@
+/*
```

Comment:
This header file should live under hw/mcu/native/include/mcu.

Reply:
Just moved this file and renamed it, but should I rename all of the functions as well to start with `mcu_sim_i2c` rather than the current `hal_i2c_sim`? The current function names now conflict with the file name, BUT I personally think that it's useful to make it clear that the functions here are implementing hal_i2c signatures. I'll follow your lead if you want me to change all the names to match the filename though.